Feature/add boost

KJonline opened the pull request:
**Description:**
Add boost services to HA and documentation has been updated with service examples and information.

**Pull request in home-assistant (if applicable):** home-assistant/home-assistant#26789

## Checklist:

- [X] Branch: `next` is for changes and new documentation that will go public with the next Home Assistant release. Fixes, changes and adjustments for the current release should be created against `current`.
- [X] The documentation follows the [standards][standards].

[standards]: https://developers.home-assistant.io/docs/documentation_standards.html

Commit:
Added Hive Water Heater

Commit:
Updated Climate

Commit:
Merge branch 'next' of https://github.com/home-assistant/home-assistant.io into next

* 'next' of https://github.com/home-assistant/home-assistant.io: (1574 commits)
  Add Twente Milieu integration (#9851)
  Add climate services to Homematic IP Cloud (#9832)
  Add documentation for Splunk Component entity filter support (#9831)
  Added config for multiple accounts - N26 (#9577)
  Add Apache Kafka docs (#9836)
  Remove monitored conditions from syncthru (#9825)
  Cleans up front matter (#9835)
  Add WWLLN docs (#9802)
  [Toon] New scan_interval on integration level (#9488)
  Update alexa.smart_home.markdown (#9827)
  Clarify
  media_player volume control (#9829)
  Add marker sensor to CUPS (#9820)
  Updates climate platform docs for Climate 1.0
  Updates Google Assistant for Climate 1.0
  Updates generic_thermostat for Climate 1.0
  Updates KNX for Climate 1.0
  Updates Stiebel Eltron to Climate 1.0
  Updates Ecobee for Climate 1.0
  Updates Radio Thermostat for Climate 1.0
  ...

# Conflicts:
#	source/_components/climate.hive.markdown

Commit:
Merged into one hive.markdown

Commit:
removed _

Commit:
Capitalised

frenck committed:
:pencil2: Tweak

Commit:
Merge branch 'next' of https://github.com/home-assistant/home-assistant.io into next

* 'next' of https://github.com/home-assistant/home-assistant.io: (66 commits)
  Update totalconnect.markdown (#9589)
  add HmIP-FCI1, HmIP-BBL, HmIP-FBL to Homematic IP Cloud (#9873)
  Update neato configuration to include vendor support (#9882)
  Add documentation for new Rainforest Eagle-200 platform (#9761)
  Add MQTT climate precision (#9896)
  Adds config validation to configuration plugin (#9886)
  Fix example (#9899)
  Clarify modes is limited to a specific set. (#9897)
  Mode `on` is now `heat` (#9893)
  Update climate.markdown (#9894)
  Collection of configuration block fixes (#9892)
  Collection of configuration block fixes (#9884)
  Add info about concurrent use of data and data_template (#9889)
  Add info about concurrent use of data and data_template (#9890)
  Updating template example (#9891)
  Mention podcast
  Mention 25k stars
  Update solaredge_local.markdown (#9878)
  Add Py35 msg
  Fix code blocks in breaking changes
  ...

Commit:
Merge branch 'next' of github.com:Rendili/home-assistant.github.io into next

Commit:
Add Boost Services Docs

Commit:
Add Boost Services Docs

Commit:
Updated Hive Docs

Commit:
Updated Hive Docs

Commit:
Updated hotwater

Commit:
Updated hotwater

Commit:
Merge branch 'next' of github.com:home-assistant/home-assistant.io into next

* 'next' of github.com:home-assistant/home-assistant.io: (46 commits)
  Document port number (#10385)
  Update Plex for config flow (#10389)
  Add Home Assistant Cast service docs (#10339)
  Update changelog.markdown (#10344)
  Add options to badges (#10286)
  Add icon_code sensor (#10365)
  List of expected sensors and their states (#10361)
  Enphase envoy Add optional name field (#10378)
  Change nzbget to full component (#10300)
  Clarify Plex config options (#10352)
  Document binary_sensor platfor for iaqualink integration (#10354)
  Remove references to legacy configuration method (#10345)
  Update ha_release (#10346)
  Added doods component documentation (#10228)
  Fix NM (#10341)
  Plex config changes (#10342)
  Add Obihai Integration (#10324)
  Document switch platform for iaqualink integration (#10335)
  Add password for upc connect (#10340)
  Plex common config (#10328)
  ...

Commit:
Merge branch 'next' into feature/add_boost

* next: (46 commits)
  Document port number (#10385)
  Update Plex for config flow (#10389)
  Add Home Assistant Cast service docs (#10339)
  Update changelog.markdown (#10344)
  Add options to badges (#10286)
  Add icon_code sensor (#10365)
  List of expected sensors and their states (#10361)
  Enphase envoy Add optional name field (#10378)
  Change nzbget to full component (#10300)
  Clarify Plex config options (#10352)
  Document binary_sensor platfor for iaqualink integration (#10354)
  Remove references to legacy configuration method (#10345)
  Update ha_release (#10346)
  Added doods component documentation (#10228)
  Fix NM (#10341)
  Plex config changes (#10342)
  Add Obihai Integration (#10324)
  Document switch platform for iaqualink integration (#10335)
  Add password for upc connect (#10340)
  Plex common config (#10328)
  ...

klaasnicolaas commented:
I think there are many commits from an older PR 🤔

KJonline commented:
> I think there are many commits from an older PR 🤔

@klaasnicolaas why would old commits pull through from my previous PR? Do I need to do anything with them and how could I fix it if needed. I am a bit of  a beginner with GIT so I am not sure how to remove the old commits from this PR.

Commit:
Updated based on code review

Commit:
Updated based on code review

MartinHjelmare commented:
Before you branch off and create a new feature branch, make sure to pull the latest version of the target branch, eg `next` in this case, from upstream remote repository.

KJonline commented:
@MartinHjelmare  if I rebased from dev. Would this correct the problem?

MartinHjelmare commented:
Probably yes.

Commit:
Merge branch 'next' of github.com:home-assistant/home-assistant.io into next

* 'next' of github.com:home-assistant/home-assistant.io: (82 commits)
  Added more ebusd sensors. (#10398)
  Describe config options (#10450)
  Update ecobee to include config flow instructions (#10428)
  Remove deprecated fedex integration (ADR-0004) (#10420)
  Remove deprecated usps integration (ADR-0004) (#10422)
  Expand entity-filter options (#10349)
  Remove deprecated ups integration (ADR-0004) (#10423)
  Added 'availability_template' to Template Sensor platform (#10308)
  Add documentation for here_travel_time (#9648)
  Add Alarm Control Panel to secure devices (#10250)
  Remove deprecated linksys_ap integration (ADR-0004) (#10434)
  Clarity on emabling MQTT TLS (#10453)
  Update vesync.markdown (#10444)
  Adds key codes to document (#10445)
  Add details about iFrame (#10441)
  Update daikin.markdown (#10439)
  Missing end bracket of `is_state` function (#10440)
  add url action (#10413)
  Update homematic.markdown
  Remove deprecated sytadin integration (ADR-0004) (#10419)
  ...

Commit:
Merge branch 'feature/add_boost' of github.com:KJonline/home-assistant.io into feature/add_boost

* 'feature/add_boost' of github.com:KJonline/home-assistant.io:
  Updated based on code review
  Updated hotwater
  Updated Hive Docs
  Add Boost Services Docs

KJonline commented:
Hi @MartinHjelmare, I have rebased the feature branch from the next branch now.

MartinHjelmare commented:
There are still old commits in the branch. Just merging next into your branch won't remove the old commits. You need to rebase and rewrite the history of this branch.

frenck commented on source/_components/hive.markdown:
```diff
@@ -9,28 +9,33 @@ ha_category:
 redirect_from:
   - /components/binary_sensor.hive/
   - /components/climate.hive/
+  - /components/water_heater.hive/
```
There is no need to create a redirect for this, since the page has not existed before.
Please remove this line.

frenck commented on source/_components/hive.markdown:
```diff
@@ -9,28 +9,33 @@ ha_category:
 ---
 
-The `hive` integration is the main integration to set up and integrate all supported Hive devices. Once configured with the minimum required details it will detect and add all your Hive devices into Home Assistant, including support for multizone heating.
+The `hive` integration is the main integration to set up and integrate all supported Hive devices. Once configured with the minimum required details it will detect and add all    Hive devices into Home Assistant, including support for multi zone heating.
```
```suggestion
The `hive` integration is the main integration to set up and integrate all supported Hive devices. Once configured with the minimum required details it will detect and add all Hive devices into Home Assistant, including support for multi-zone heating.
```

frenck commented on source/_components/hive.markdown:
```diff
@@ -9,28 +9,33 @@ ha_category:
+The `hive` integration is the main integration to set up and integrate all supported Hive devices. Once configured with the minimum required details it will detect and add all    Hive devices into Home Assistant, including support for multi zone heating.
 
-This integration uses the unofficial API used in the official Hive website [https://my.hivehome.com](https://my.hivehome.com), and you will need to use the same Username and Password you use on the Hive website to configure this Hive integration in Home Assistant.
+This integration uses the unofficial API used in the official Hive website [https://my.hivehome.com](https://my.hivehome.com), and you will need to use the same Username and P    you use on the Hive website to configure this Hive integration in Home Assistant.
```
What happened to "Password" -> "P" ?

frenck commented on source/_components/hive.markdown:
```diff
@@ -57,6 +62,62 @@ scan_interval:
   default: 2
 {% endconfiguration %}
 
+## Service - Boost Heating
```
```suggestion
### Service `hive.boost_heating`
```

frenck commented on source/_components/hive.markdown:
````diff
@@ -57,6 +62,62 @@ scan_interval:
+          temperature: "20.5"
+```
+
+## Service - Boost Hotwater
````
```suggestion
### Service `hive.boost_hotwater`
```

frenck commented on source/_components/hive.markdown:
```diff
@@ -57,6 +62,62 @@ scan_interval:
+
+## Service - Boost Hotwater
+
+You can use the service `hive.boost.hotwater` to set your hot water to boost for a period of time.
```
```suggestion
You can use the service `hive.boost_hotwater` to set your hot water to boost for a period of time.
```

frenck commented on source/_components/hive.markdown:
```diff
@@ -57,6 +62,62 @@ scan_interval:
+| Service data attribute | Optional | Description                                                            |
+| ---------------------- | -------- | ---------------------------------------------------------------------- |
+| `entity_id`            | no       | String, Name of entity e.g. `water_heater.hot_water`.                  |
+| `time_period`          | yes      | Time Period, Period of time the boost should last for e.g. `01:30:00`. |
```
```suggestion
| `time_period`          | yes      | Time Period, Period of time the boost should last for e.g., `01:30:00`. |
```

frenck commented on source/_components/hive.markdown:
```diff
@@ -57,6 +62,62 @@ scan_interval:
+script:
+  boost_hotwater:
+    sequence:
+      - service: "hive.boost_hotwater"
```
`hotwater` is actually not a valid dictionary word, `hot water` is. I would have expected the service to be `hive.boost_hot_water`.

frenck commented on source/_components/hive.markdown:
```diff
@@ -57,6 +62,62 @@ scan_interval:
+
+<div class='note'>
+
+You must have the [Hive component](/components/hive/) configured to use the platforms below.
```
```suggestion
You must have the [Hive integration](/components/hive/) configured to use the platforms below.
```

frenck commented on source/_components/hive.markdown:
```diff
@@ -75,7 +136,7 @@ A short boost for Hive Heating can be set by using the **Boost** preset, this wi
 
 - Hive Active Heating
-- Hive Multizone
+- Hive Multi zone
```
```suggestion
- Hive Multi-zone
```

frenck commented on source/_components/hive.markdown:
```diff
@@ -57,6 +62,62 @@ scan_interval:
+| ---------------------- | -------- | ---------------------------------------------------------------------- |
+| `entity_id`            | no       | String, Name of entity e.g. `climate.heating`.                         |
+| `time_period`          | no       | Time Period, Period of time the boost should last for e.g. `01:30:00`. |
+| `temperature`          | yes      | String, The required target temperature e.g. `20.5`                    |
```
```suggestion
| `temperature`          | yes      | String, The required target temperature e.g., `20.5`                    |
```

frenck commented on source/_components/hive.markdown:
```diff
@@ -57,6 +62,62 @@ scan_interval:
+| Service data attribute | Optional | Description                                                            |
+| ---------------------- | -------- | ---------------------------------------------------------------------- |
+| `entity_id`            | no       | String, Name of entity e.g. `climate.heating`.                         |
+| `time_period`          | no       | Time Period, Period of time the boost should last for e.g. `01:30:00`. |
```
```suggestion
| `time_period`          | no       | Time Period, Period of time the boost should last for e.g., `01:30:00`. |
```

frenck commented on source/_components/hive.markdown:
```diff
@@ -57,6 +62,62 @@ scan_interval:
   default: 2
 {% endconfiguration %}
 
```
Add a new chapter here: `## Services`

frenck commented on source/_components/hive.markdown:
```diff
@@ -9,28 +9,33 @@ ha_category:
-- [Sensor](#Sensor)
-- [Switch](#Switch)
-- [Water Heater](#Water-Heater)
+- [Service - Boost Heating](#service---boost-heating)
```
```suggestion
```

This is a list of supported platforms, not a TOC.

frenck commented on source/_components/hive.markdown:
```diff
@@ -9,28 +9,33 @@ ha_category:
-- [Switch](#Switch)
-- [Water Heater](#Water-Heater)
+- [Service - Boost Heating](#service---boost-heating)
+- [Service - Boost Hotwater](#service---boost-hotwater)
```
```suggestion
```

This is a list of supported platforms, not a TOC.

frenck reviewed (changes requested):
Thanks for the PR @KJonline! I've reviewed it and left you some comments. Could you please take a look? Thanks! 👍

Commit:
Update source/_components/hive.markdown

Co-Authored-By: Franck Nijhof <frenck@frenck.nl>

Commit:
Update source/_components/hive.markdown

Co-Authored-By: Franck Nijhof <frenck@frenck.nl>

Commit:
Update source/_components/hive.markdown

Co-Authored-By: Franck Nijhof <frenck@frenck.nl>

Commit:
Update source/_components/hive.markdown

Co-Authored-By: Franck Nijhof <frenck@frenck.nl>

Commit:
Update source/_components/hive.markdown

Co-Authored-By: Franck Nijhof <frenck@frenck.nl>

Commit:
Update source/_components/hive.markdown

Co-Authored-By: Franck Nijhof <frenck@frenck.nl>

Commit:
Update source/_components/hive.markdown

Co-Authored-By: Franck Nijhof <frenck@frenck.nl>

Commit:
Apply suggestions from code review

Co-Authored-By: Franck Nijhof <frenck@frenck.nl>

Commit:
Review 1

Commit:
Merge branch 'feature/add_boost' of github.com:Rendili/home-assistant.github.io into feature/add_boost

# Conflicts:
#	source/_components/hive.markdown

KJonline commented:
Hi @frenck, I have updated files as per your changes. on the point about hotwater and hot water. I have updated the docs and created a second PR updating the in the code to reflect the changes in there also.

@MartinHjelmare , I did a rebase on this branch the merge its referring to is from a couple of weeks ago. Just so you are aware as above I have created a new PR to change hotwater to hot_water in the home assistant code. Please could you review this as I know the other PR has been merged and closed. The new one I created is home-assistant/home-assistant#27032

MartinHjelmare commented:
You can do a rebase in more than one way. Here we need to remove the old commits during the rebase. If you do an interactive rebase you can select what commits to keep, what to drop and what order they should go.
